packages/server/src/api/muster/muster.controller.ts

```diff
@@ -233,22 +237,110 @@ class MusterController {
   }
 
   /**
-   * This method returns the normalized rate (0 - 1.0) of compliance on a given
-   * date for a given unit, against all of the unit's muster configs.
+   * Provides Muster Compliance details for all service members for the given unit id(s) curent roster
    */
-  async getMusterComplianceByDate(
-    req: ApiRequest<{ orgId: number; unitName: string }, null, GetMusterComplianceByDateQuery>,
-    res: Response<GetMusterComplianceByDateResponse>,
-  ) {
-    const musterComplianceRate = await getMusterComplianceByDate(req.params.orgId, req.params.unitName, req.query.isoDate);
-    res.json({ musterComplianceRate, isoDate: req.query.isoDate });
+  async getRosterMusterComplianceByDateRange(req: ApiRequest<OrgRoleParams, null, GetMusterRosterQuery>, res: Response<Paginated<Partial<MusterCompliance>>>) {
+    assertRequestQuery(req, ['fromDate', 'toDate', 'limit', 'page']);
+
+    /* Dates come in UTC timezone.
+      For example the date 2021-07-04 11:59 PM selected in a browser
+                  comes as 2021-07-05T04:59:00.000Z */
+    const fromDate: moment.Moment = moment(req.query.fromDate, 'YYYY-MM-DD');
+    const toDate: moment.Moment = moment(req.query.toDate, 'YYYY-MM-DD');
+
+    if (!fromDate.isValid() || !toDate.isValid() || fromDate > toDate) {
+      throw new BadRequestError('Invalid ISO date range.');
+    }
+
+    const rowLimit = parseInt(req.query.limit);
+    const pageNumber = parseInt(req.query.page);
+    // When unit id is missing, then data for all units are requested
+    const unitId = req.query.unitId!;
+    const orgId = req.appOrg!.id;
+
+    Log.info(`getUserMusterCompliance() for ${JSON.stringify({ orgId, unitId, fromDate, toDate })}`);
+
+    // get units for the org
+    const orgUnits = await Unit.createQueryBuilder('unit')
+      .select()
+      .where('unit.org_id = :orgId', { orgId })
+      .getMany();
```

Comment:
This style should be a little less error prone, and a little more readable I think. I generally only use the query builder as a fallback when something more advanced needs to happen.

```typescript
const orgUnits = await Unit.find({
  where: { org: orgId },
});
```

There is one bug with this style that Peter ran into, where the `Between` function doesn't seem to work correctly with timestamps. It looked like there was an open issue for it. Pretty lame, but other than that it should work well.